src/components/PhoneField/PhoneField.tsx

```diff
@@ -0,0 +1,338 @@
+import React, {useState, useEffect, useCallback} from 'react';
+
+import {TextField} from '../TextField';
+import {Button} from '../Button';
+import {Popover} from '../Popover';
+import {ActionList} from '../ActionList';
+
+import styles from './PhoneField.scss';
+
+export interface Country {
+  /** The country flag */
+  image: string;
+  /** Country name */
+  countryName: string;
+  /** Country area code */
+  countryCode: string;
+  countryAlphaCode?: string;
+  /** Phone number display format */
+  displayFormat: number[];
+  /** Possible area codes for a country. Used to distinguish between countries with same country codes */
+  areaCodes?: number[];
+  population?: number;
+  formatter?(): void;
+}
+
+export interface PhoneFieldProps {
+  /** Placeholder text  */
+  placeholder?: string;
+  /** Textfield label  */
+  labelName?: string;
+  /** Hides the label for accessibility */
+  labelHidden?: boolean;
+  /** Is textfield optional */
+  optional?: boolean;
```

Comment:
I don't think this adds much value - if people want to add "optional" text at the end of their string then they should add it in the labelName. Also this kind of string concatenation doesn;t really work with internationalisation - some we'd need to translate the "optional" and there's not guarentee its location will be in a meaningfull place in non-english languages

Reply:
Be careful about auto-constructing labels by piecing strings together.  Depending on the language, `(optional)` may need to adjust, e.g. in gender, to match what it's modifying.